test/unicode/checkstring.jl

```diff
@@ -89,11 +89,7 @@ try
         @test_throws UnicodeError Base.checkstring(UInt8[byt,0x80,0x80,0xc0])
     end
 
-    # Long encoding of 0x01
-    @test_throws UnicodeError utf8(b"\xf0\x80\x80\x80")
-    # Test ends of long encoded surrogates
-    @test_throws UnicodeError utf8(b"\xf0\x8d\xa0\x80")
-    @test_throws UnicodeError utf8(b"\xf0\x8d\xbf\xbf")
```

Comment:
At least the data being tested here is useful, so ok if I add these back under `convert` for now? The same data can be tested under whatever the new design ends up being.